test/test.js

```diff
@@ -6,75 +6,238 @@ const test = require('ava')
+
   const paymentClient = new hipstershop.PaymentService(`0.0.0.0:${process.env.PAYMENT_SERVICE_PORT}`, grpc.credentials.createInsecure())
   charge = promisify(paymentClient.charge).bind(paymentClient)
 
+  const productCatalogClient = new hipstershop.ProductCatalogService(`0.0.0.0:${process.env.PRODUCT_CATALOG_SERVICE_PORT}`, grpc.credentials.createInsecure())
+  productList = promisify(productCatalogClient.listProducts).bind(productCatalogClient)
+  productGet = promisify(productCatalogClient.getProduct).bind(productCatalogClient)
+  productSearch = promisify(productCatalogClient.searchProducts).bind(productCatalogClient)
+
   const recommendationClient = new hipstershop.RecommendationService(`0.0.0.0:${process.env.RECOMMENDATION_SERVICE_PORT}`, grpc.credentials.createInsecure())
   recommend = promisify(recommendationClient.listRecommendations).bind(recommendationClient)
+
+  const shippingClient = new hipstershop.ShippingService(`0.0.0.0:${process.env.SHIPPING_SERVICE_PORT}`, grpc.credentials.createInsecure())
+  shippingQuote = promisify(shippingClient.getQuote).bind(shippingClient)
+  shippingOrder = promisify(shippingClient.shipOrder).bind(shippingClient)
+})
+
+// --------------- Ad Service ---------------
+
+test('ad: get', async t => {
+  const req = data.ad
+  const res = await adsGet(req)
+
+  t.is(res.ads.length, 2)
+  t.truthy(res.ads[0].redirectUrl)
+  t.truthy(res.ads[1].redirectUrl)
+  t.truthy(res.ads[0].text)
+  t.truthy(res.ads[1].text)
+})
+
+// --------------- Cart Service ---------------
+
+test('cart: all', async t => {
+  const req = data.cart
+  const userIdReq = { userId: req.userId }
+
+  // Empty Cart
+  let res = await cartEmpty(userIdReq)
+  t.truthy(isEmpty(res))
+
+  // Add to Cart
+  res = await cartAdd(req)
+  t.truthy(isEmpty(res))
+
+  // Check Cart Content
+  res = await cartGet(userIdReq)
+  t.is(res.items.length, 1)
+  t.is(res.items[0].productId, req.item.productId)
+  t.is(res.items[0].quantity, req.item.quantity)
+
+  // Empty Cart
+  res = await cartEmpty(userIdReq)
+  t.truthy(isEmpty(res))
+
+  // Check Cart Content
+  res = await cartGet(userIdReq)
+  t.truthy(isEmpty(res))
+})
+
+// --------------- Currency Service ---------------
+
+test('currency: supported', async t => {
+  const res = await currencySupported({})
+  t.is(res.currencyCodes.length, 33)
+})
+
+test('currency: convert', async t => {
+  const req = data.currency
+
+  const res = await currencyConvert(req)
+  t.is(res.currencyCode, "CAD")
+  t.is(res.units, 442)
+  t.is(res.nanos, 599380805)
+})
+
+// --------------- Checkout Service ---------------
+
+test('checkout: place order', async t => {
+  const req = data.checkout
+  const res = await checkoutOrder(req)
+
+  t.truthy(res.order.orderId)
+  t.truthy(res.order.shippingTrackingId)
+  t.truthy(res.order.shippingAddress)
+  t.is(res.order.shippingCost.currencyCode, 'USD')
+})
+
+// --------------- Email Service ---------------
+
+// TODO
```

Comment:
This line maybe redundant?

Reply:
this `TODO` is in reference to the email service below. It is a bit different than all the others since it works with JSON POST requests instead of using gRPC service loaded from the protobuf.